index.rst

```diff
@@ -592,6 +594,99 @@ Parsing XML Files
 The Kafka version of ts_salobj parses ts_xml topic schema files directly on the fly, instead of parsing OMG IDL files as the DDS version does.
 This turns out to be very fast; tests/test_speed.py on a mac report: "Created 859.3 topic classes/sec (129 topic classes); total duration 0.15 seconds" when parsing topic classes for MTM1M3, a component which has an unusually large number of topics, including several unusually large topics.
 
+Topic Partitioning
+------------------
+
+While both DDS and Kafka provide the concept of topic partitions, their purposes and usage differ significantly: DDS uses partitions for isolating topics, whereas Kafka uses them to optimize performance and scalability.
+
+In DDS, partitions have text-based names and can be freely created while producing or consuming topics.
+They are typically used to isolate topics, organizing them into logical groups.
+
+In Kafka, partitions are numbered, and the number of partitions for a topic must be preconfigured on the brokers.
+Increasing the number of partitions is possible, but reducing it is not straightforward.
+Additionally, it is possible to control which partition a topic is written to by using keys.
+In Kafka, partitions serve primarily to enhance system throughput by enabling parallelism.
+
+Another key aspect of topic partition in Kafka is that to ensure ordered delivery, topics need to be in the same partition.
+This is critical for CSC commands, especially in time-critical operations like tracking.
+However, since commands rely on command/acknowledgement procedure, it is hard to envision situations where commands from the same source would be delivered out of order.
+Furthermore, it is rare that more than one client issues commands to the same CSC and, in such cases, delivery order is probably not going to be critical.
+In any case, it is not advisable to partition command topics.
+
+For events and telemetry topics, the situation is a bit more nuanced.
+
+Event topics, on the other hand, are asynchronously published by components to indicate state changes.
```

Comment:
I would drop `on the other hand` here since from the last sentence it was already clear that event topics (and telemetry topics) would be discussed next and telemetry topics have not been discussed yet.